### ISSUE TITLE
aws_launch_template ignores device_index=0

### ISSUE THREAD

**Issue by joernheissler** [terraform-providers/terraform-provider-aws#4475]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.11.6
+ provider.aws v1.17.0
+ provider.external v1.0.0
+ provider.local v1.1.0
+ provider.template v1.0.0
```

### Affected Resource(s)

* `aws_launch_template`

### Terraform Configuration Files

```hcl
resource "aws_launch_template" "loadtest" {
    name = "loadtest"
    description = "LoadTest"
    image_id = "${data.aws_ami.ubuntu_xenial.id}"
    instance_initiated_shutdown_behavior = "terminate"
    instance_type = "t2.small"
        
    placement {
        availability_zone = "eu-central-1a"
    }       
            
    network_interfaces {
        subnet_id = "${element(aws_subnet.stratus_private.*.id, 0)}"
# ----------------- Problem is here:
        device_index = 0
# -----------------
        delete_on_termination = true
        associate_public_ip_address = false
        security_groups = [
            "${aws_security_group.loadtest.id}",
        ]
    }
}
```

### Expected Behavior

Terraform should submit the device index (0) to AWS.

### Actual Behavior

Terraform ignores zero device index. AWS will then complain:
`You must use a valid fully formed launch template. Each network interface requires a device index.`

### Steps to Reproduce

1. Delete default VPC
2. Create new (non-default) VPC, subnets, etc. etc. etc.
3. Use above resource and an autoscale group which uses the launch template.

### Important Factoids

By deleting the default VPC, AWS will complain `No default VPC for this user`. So I specified `network_interfaces` like above and got the problem described above.

I quick fixed the issue in `aws/resource_aws_launch_template.go`:
```go
    if v, ok := ni["device_index"].(int); ok && v != 0 {
        networkInterface.DeviceIndex = aws.Int64(int64(v))
    }
```
I removed `&& v != 0` and when I first destroyed the resource, it got created correctly on next apply. But updates don't work, probably another comparison that needs fixing.

**Comment by bflad** [comment]
FYI there's already a PR associated with this: #4367

**Comment by bflad** [comment]
The simple fix of removing the `v != 0` check has been merged into master via #4367 and will release with v1.19.0 of the AWS provider, likely middle of next week. If there are other continuing issues after that, please open a new issue. Thanks!
